app/models/mention.rb

```diff
@@ -5,16 +5,22 @@
 class Mention < ActiveRecord::Base
   REGEX = /@\{([^;]+); ([^\}]+)\}/
 
-  belongs_to :post
+  belongs_to :mentions_container, polymorphic: :true
+  belongs_to :post, ->(mention) { (mention.mentions_container_type == "Post") ? where("1 = 1") : none },
```

Comment:
I picked the idea from the second answer [here](http://stackoverflow.com/questions/21850995/association-for-polymorphic-belongs-to-of-a-particular-type). I don't like the self-reference approach from the first answer, it looks less-obvious for me.